src/common/image_cache.c

```diff
@@ -178,16 +155,23 @@ void dt_image_cache_print(dt_image_cache_t *cache)
          (float)cache->cache.cost / (float)cache->cache.cost_quota);
 }
 
-const dt_image_t *dt_image_cache_read_get(dt_image_cache_t *cache, const uint32_t imgid)
+dt_image_t *dt_image_cache_get(dt_image_cache_t *cache, const uint32_t imgid, char mode)
 {
   if(imgid <= 0) return NULL;
```

Comment:
what about the == 0 case? i thought sqlite indices were 1-based. anyways this is for the case where the cache doesn't return an image because it is full and all entries are locked/can't be evicted. i don't think this should happen any more (the new cache will temporarily go over memory budget to avoid this.. of course crashing if the caller requests an unbounded amount of data and never releases the locks).